### ISSUE TITLE
Export to Onnx

### ISSUE THREAD

**Issue by marcojoao** [saic-mdal/lama#84]
Got this failed attempt to convert the model to ONNX

Code example:
save_onnx_path = "/content/lama.onnx"
```
image = torch.rand(1, 3, 120, 120)
mask = torch.rand(1, 1, 120, 120)
inputs = {
    "image": img,
    "mask": mask
}

torch.onnx.export(model,   
                  inputs,   
                  save_onnx_path,
                  opset_version=12,
                  do_constant_folding=True,
                  input_names = ['img', 'mask'],
                  output_names = ['output'],
                  dynamic_axes={
                      'img' : {0 : 'batch_size', 2 : 'width', 3 : 'height'},
                      'mask' : {0 : 'batch_size', 2 : 'width', 3 : 'height'},
                      'output' : {0 : 'batch_size', 2 : 'width', 3 : 'height'},
                  })
```

Stack trace:
`RuntimeError: Exporting the operator fft_rfftn to ONNX opset version 12 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.`

Did anyone know where is this "fft_rfftn" to find a possible solution.

**Comment by windj007** [comment]
Hi! Yeah, it seems that ONNX [does not support FFT/IFFT](https://github.com/onnx/onnx/blob/main/docs/Operators.md)

**Comment by windj007** [comment]
Feel free to reopen the issue if you heave further questions.

**Comment by q671383789** [comment]
@marcojoao Hi, bro. I'm facing the same question, while trying to use torch.onnx to export lama with rfft,
Have u sovled this?

**Comment by ssometimes** [comment]
> @marcojoao Hi, bro. I'm facing the same question, while trying to use torch.onnx to export lama with rfft, Have u sovled this?

I also encountered the same error, did you solve it?😂

**Comment by marcojoao** [comment]
> @marcojoao Hi, bro. I'm facing the same question, while trying to use torch.onnx to export lama with rfft, Have u sovled this?

> > @marcojoao Hi, bro. I'm facing the same question, while trying to use torch.onnx to export lama with rfft, Have u sovled this?
> 
> I also encountered the same error, did you solve it?😂

Hello, short answer no, I did notice that Onnx team already included the necessary function, but pytorch did not update to the latest Onnx version, so here is no conversion for this new function.

I successfully converted the the fallback to "Atan" but the results are not that great.
If you guys dont wanna wait, a possible solution is to compile pytorch with the new Onnx version and your custom C++ wrapper to the new function, if you are brave enougth.

**Comment by q671383789** [comment]
@marcojoao  Thx a lot. That's a good way to do it, but seems like really hard. So I will bailan(means wait) till pytorch officionally update with onnx. 0u0

**Comment by ulfemsoy** [comment]
> > @marcojoao Hi, bro. I'm facing the same question, while trying to use torch.onnx to export lama with rfft, Have u sovled this?
> 
> > > @marcojoao Hi, bro. I'm facing the same question, while trying to use torch.onnx to export lama with rfft, Have u sovled this?
> > 
> > 
> > I also encountered the same error, did you solve it?😂
> 
> Hello, short answer no, I did notice that Onnx team already included the necessary function, but pytorch did not update to the latest Onnx version, so here is no conversion for this new function.
> 
> I successfully converted the the fallback to "Atan" but the results are not that great. If you guys dont wanna wait, a possible solution is to compile pytorch with the new Onnx version and your custom C++ wrapper to the new function, if you are brave enougth.

@marcojoao, do you know if the conversion to .onnx is possible now, one year later?

**Comment by AMEERAZAM08** [comment]
Hii all can i get lama.onnx file. ? @windj007 @marcojoao @wcyjerry @ulfemsoy @ssometimes

**Comment by LLSean** [comment]
Has anyone successfully converted so far?

**Comment by ulfemsoy** [comment]
No, not any success so far. Seems to be caused by still missing functionality in ONNX compared to PyTorch.

**Comment by OPHoperHPO** [comment]
Hello everyone,

We at @Carve-Photos have successfully ported LaMa to ONNX with results closely resembling the original. Check it out here: https://huggingface.co/Carve/LaMa-ONNX

**Comment by WeiboXu** [comment]
> Hello everyone,
> 
> We at @Carve-Photos have successfully ported LaMa to ONNX with results closely resembling the original. Check it out here: https://huggingface.co/Carve/LaMa-ONNX

when set 
dynamic_axes={
                      'img' : {0 : 'batch_size', 2 : 'width', 3 : 'height'},
                      'mask' : {0 : 'batch_size', 2 : 'width', 3 : 'height'},
                      'output' : {0 : 'batch_size', 2 : 'width', 3 : 'height'},
                  }
How to "Adapt FourierUnit to support dynamic axes (see irfttn and rfft for correct padding)"?
Thanks.
